admin-dev/themes/new-theme/js/pages/product/edit/product-partial-updater.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+const {$} = window;
+
+export default class ProductPartialUpdater {
+  constructor($productForm, $productFormSubmitButton) {
+    this.$productForm = $productForm;
+    this.$productFormSubmitButton = $productFormSubmitButton;
+  }
+
+  watch() {
+    this.$productFormSubmitButton.prop('disabled', true);
+    this.initialData = this.getFormDataAsObject();
+    this.$productForm.submit((e) => this.updatePartialForm(e));
+    this.$productForm.on('change', ':input', () => this.updateSubmitState());
+  }
+
+  updatePartialForm(event) {
+    event.stopImmediatePropagation();
+
+    const updatedData = this.getUpdatedFormData();
+    if (updatedData !== null) {
+      this.postUpdatedData(updatedData);
+    } else {
+      alert('no fields updated');
```

Comment:
An alert ? with a text no translated ?

Reply:
It is temporary, I don't know how we will handle it (modal, session message, ...)

Actually currently the save button is disabled as long as no updated data has been edited so it shouldn't even be called anyway But keep in mind it's not definitive so it shouldn't be blocking for this PR 🙂